lib/tokenize.cpp

```diff
@@ -11376,12 +11376,27 @@ void Tokenizer::simplifyAsm()
         }
 
         else if (Token::Match(tok, "asm|__asm|__asm__ volatile|__volatile|__volatile__| (")) {
+            // Searching for unmatched '{' before asm (label or inline?)
+            const Token* uptok = tok->previous();
+            while (uptok && uptok->str() != "{") {
```

Comment:
Spontanously it seems to me that this loop should be more strict. 
Should you continue looping if there is a `;` for instance?

I don't see what test cases this handle. So I don't spontanously feel that a `}` should be skipped like this.